internal/api/edit_task.go

```diff
@@ -0,0 +1,152 @@
+package api
+
+import (
+	"database/sql"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"go_final_project/internal/config"
+	"go_final_project/internal/db"
+	"go_final_project/internal/scheduler"
+)
+
+// GetTaskHandler обрабатывает GET-запрос для получения задачи по идентификатору.
+func GetTaskHandler(w http.ResponseWriter, r *http.Request) {
+	taskID := r.URL.Query().Get("id")
+
+	// Проверка, указан ли идентификатор
+	if taskID == "" {
+		http.Error(w, `{"error": "Не указан идентификатор"}`, http.StatusBadRequest)
+		return
+	}
+
+	// Подключение к базе данных
+	dbConn, err := db.SetupDB()
+	if err != nil {
+		http.Error(w, `{"error": "ошибка подключения к базе данных"}`, http.StatusInternalServerError)
+		return
+	}
+	defer dbConn.Close()
+
+	// Получение задачи по идентификатору
+	var task config.Task
+	query := `SELECT id, date, title, comment, repeat FROM scheduler WHERE id = ?`
+	err = dbConn.Get(&task, query, taskID)
+
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Ошибку лучше сравнивать так 
```suggestion
		if errors.Is(err, sql.ErrNoRows) {
```